aptos-move/aptos-vm/src/block_executor/vm_wrapper.rs

```diff
@@ -45,15 +45,10 @@ impl<'a, S: 'a + StateView + Sync> ExecutorTask for AptosExecutorTask<'a, S> {
         txn_idx: TxnIndex,
         materialize_deltas: bool,
     ) -> ExecutionStatus<AptosTransactionOutput, VMStatus> {
-        // TODO[agg_v2](fix) look at whether it is enabled, not just capable.
-        // if it is disabled, no need to fallback.
-        if txn.is_valid() && executor_with_group_view.is_delayed_field_optimization_capable() {
-            if let Transaction::GenesisTransaction(WriteSetPayload::Direct(_)) = txn.expect_valid()
-            {
-                // WriteSetPayload::Direct cannot be handled in mode where delayed_field_optimization is enabled
-                // And we need to communicate to the BlockExecutor, so they can retry with capability disabled
-                return ExecutionStatus::DirectWriteSetTransactionNotCapableError;
-            }
+        if executor_with_group_view.is_delayed_field_optimization_capable()
```

Comment:
is there any usecase now for having is_delayed_field_optimization_capable separate?

Reply:
yes, executing transactions outside of BlockExecutor context are not capable of handling delayed fields, and so we need to fall back.

similarly, within BlockExecutor, if we've received a single Direct WriteSet transaction, we need to set capable=false in the resolver.